packages/components/src/responsive-toolbar/index.js

```diff
@@ -0,0 +1,158 @@
+			countHiddenChildren: 0,
+		};
+		this.container = createRef();
+		this.hiddenContainer = createRef();
+
+		this.throttledUpdateHiddenItems = this.throttledUpdateHiddenItems.bind( this );
+	}
+
+	componentDidMount() {
+		this.toggleWindowEvents( true );
+		this.updateHiddenItems();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.children !== this.props.children ) {
+			this.updateHiddenItems();
+		}
+	}
+
+	componentWillUnmount() {
+		this.toggleWindowEvents( false );
+	}
+
+	toggleWindowEvents( isListening ) {
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+
+		window.cancelAnimationFrame( this.rafHandle );
+		window[ handler ]( 'resize', this.throttledUpdateHiddenItems );
+		window[ handler ]( 'scroll', this.throttledUpdateHiddenItems, true );
+	}
+
+	throttledUpdateHiddenItems() {
+		this.rafHandle = window.requestAnimationFrame( () => this.updateHiddenItems() );
+	}
+
+	updateHiddenItems() {
+		const { instanceId } = this.props;
+		const containerRect = this.container.current.getBoundingClientRect();
+		let countHiddenChildren = 0;
+		const total = this.hiddenContainer.current.childNodes.length;
+		this.hiddenContainer.current.childNodes.forEach( ( child ) => {
+			const childRect = child.getBoundingClientRect();
+			if (
+				childRect.left < containerRect.left ||
+				childRect.right > containerRect.right - OFFSET
+			) {
+				countHiddenChildren++;
+			}
+		} );
+
+		if ( countHiddenChildren !== this.state.countHiddenChildren ) {
+			this.setState( {
+				countHiddenChildren,
+			} );
+
+			if ( this.style ) {
+				this.style.parentNode.removeChild( this.style );
+			}
+			const styleNode = document.createElement( 'style' );
+			styleNode.innerHTML = `
+				#responsive-toolbar-${ instanceId } > *:nth-child(n+${ total - countHiddenChildren + 2 }):not(.components-responsive-toolbar__dropdown) {
+					display: none;
+				}
+
+				.components-responsive-toolbar__dropdown-content-${ instanceId } .components-popover__content > *:nth-child(-n+${ total - countHiddenChildren }) {
+					display: none;
+				}
+			`;
+			document.body.appendChild( styleNode );
+			this.style = styleNode;
+		}
+	}
+
+	render() {
+		const defaultRenderToggle = ( { onToggle, isOpen } ) => (
+			<IconButton
+				icon="arrow-down-alt2"
+				onClick={ onToggle }
+				aria-expanded={ isOpen }
+			/>
+		);
+		const {
+			children,
+			instanceId,
+			className,
+			extraContentClassName,
+			renderToggle = defaultRenderToggle,
+			...props
+		} = this.props;
+		const { countHiddenChildren } = this.state;
+
+		return (
+			<div
+				id={ `responsive-toolbar-${ instanceId }` }
+				className={ classnames( className, 'components-responsive-toolbar' ) }
+				ref={ this.container }
+				{ ...props }
+			>
+				<Disabled>
+					<div className="components-responsive-toolbar__compute-position" ref={ this.hiddenContainer }>
+						{ children }
+					</div>
+				</Disabled>
+
+				{ Children.map( children, ( child, index ) => {
```

Comment:
Why not `{ children }` ?

Reply:
Also, maybe it was an earlier approach you considered, but why not just _not render_ the hidden children, rather than all the work to style them as hidden?

Reply:
Notice that we're not hiding children (components), we're hiding dom nodes. We can't work on components (children) because one child (a slot) can contain multiple dom nodes. It can even be a slot containing another slot in some cases.

Reply:
> Why not `{ children }` ?

The idea is that I'm rendering the children in multiple places, I was wondering if React uses the same "instances" of components it it finds the same instance of element. But I guess it's probably not the case.